x-pack/plugins/task_manager/server/lib/task_partitioner.test.ts

```diff
@@ -67,4 +76,22 @@ describe('getPartitions()', () => {
       247, 249, 250, 252, 253, 255,
     ]);
   });
+
+  test('correctly caches the partitions on 10 second interval ', async () => {
+    const taskPartitioner = new TaskPartitioner(POD_NAME, discoveryServiceMock);
+    await taskPartitioner.getPartitions();
+    jest.advanceTimersByTime(3000);
```

Comment:
nit: Maybe you could explicitly use `CACHE_INTERVAL` in these tests to guarantee there is no update when timers advance by `something < CACHE_INTERVAL`.

4 * 3000 millisecs feels kinda arbitrary 😅

Reply:
Sure! I updated the test in this commit, https://github.com/elastic/kibana/pull/189562/commits/dd9ff57e0bd45dcdf12b4f8ea8e9e2da9d2e49da, let me know if this is not what you're thinking